src/types/WmsLayerType.ts

```diff
@@ -0,0 +1,112 @@
+import { SpatialBounds } from './mapTypes';
+
+// TODO: reuse organisation type for rasterlayer in api/rasters.ts and this file from ./types/organisationType
+export interface Organisation {
+  name: string,
+  uuid: string,
+  url: string,
+}
+
+export interface WmsLayerReceivedFromApi {
+  name: string;
+  uuid?: string,
+  slug: string;
+  description: string;
+  datasets: {slug:string}[];
+  
+  wms_url: string;
+  download_url: string;
+  legend_url: string;
+  get_feature_info_url: string;
+
+  tiled: boolean;
+  min_zoom: number;
+  max_zoom: number;
+  spatial_bounds: SpatialBounds | null;
+  options: string;
+
+  organisation: Organisation;
+  shared_with: Organisation[];
+  access_modifier: string;
+  supplier: string;
+  supplier_code: string;
+}
+
+export const wmsLayerReceivedFromApiToForm = (wmsLayer: WmsLayerReceivedFromApi): WmsLayerFormType => {
+  return { 
+      ...wmsLayer,
+      sharedWithCheckbox: wmsLayer.shared_with.length > 1? true : false,
+      organisation: wmsLayer.organisation.uuid,
+      shared_with: wmsLayer.shared_with.map(orga=>orga.uuid),
+      datasets: wmsLayer.datasets.map(dataset=>dataset.slug),
+      options: JSON.stringify(wmsLayer.options)
+    }
+}
```

Comment:
Nice that you make these functions and it seems logical to have them in the type definition file.